### PR TITLE
Update Binding cluster requirement in device type to align with the spec

### LINKED ISSUE #22207: Fix incorrect binding cluster conformance in device types
#### Problem
#22119 introduced incorrect Binding cluster bits in various device types.

For example, in the "Matter On/Off Light Switch" device type, the Binding cluster server is required per spec, but that PR made it optional.

 #### Proposed Solution
Fix things to actually follow the spec.

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* https://github.com/project-chip/connectedhomeip/pull/22119 introduced incorrect Binding cluster bits in various device types.
* Binding is defined as "required if and only if the device type is Simple and has a Client cluster",  Which means:

1. Not included if there are no Client clusters.
2. Not included if the device type is not Simple.
3. Required if the device has a required Client cluster.
4. Optional if the device has only optional Client clusters, but must be present if any of those Client clusters are present.

That is, for any given device type the "Simple & Client" expression evaluates to a specific boolean, except if the device type has optional client clusters.

* Fixes #22207

#### Change overview
Update Binding cluster requirement in device type to align with the spec

#### Testing
How was this tested? (at least one bullet point required)
* xml update only

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -1894,9 +1873,6 @@ limitations under the License.
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1897]: The All-clusters-app device (not in the spec) should have Binding server, no?

Comment (yufengwangca) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1897]: It is not defined, do we need to to enable any client cluster on the  all-clusters-app? I think might be OTA Software Update Provider.

Maybe we should left as is for flexibility. 
<include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="false">

Comment (yufengwangca) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1897]: all-cluster-app has enabled OTA Software Update Provider with client role by default, so I include Binding in all-cluster-app with default on and allow the user to turn off.

`<include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="false">`

**Comment by andy31415** [comment]
Accepted: compatibility issue, does not actually change coresdk code.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**